File: api/src/endpoint/organization/create_organization.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;
use crate::resource::OrganizationResource;

pub struct CreateOrganization;

impl Endpoint for CreateOrganization {
    const PATH: &'static str = "/organization/{org_name}";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateOrganizationRequest;
    type Response = CreateOrganizationResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateOrganizationRequest {}

pub type CreateOrganizationResponse = OrganizationResource;

File: gitdot-api-ts/src/endpoint/organization/create-organization.ts
import { z } from "zod";

import { OrganizationResource } from "../../resource";

export const CreateOrganizationRequest = z.object({});
export type CreateOrganizationRequest = z.infer<
  typeof CreateOrganizationRequest
>;

export const CreateOrganization = {
  path: "/organization/{org_name}",
  method: "POST",
  request: CreateOrganizationRequest,
  response: OrganizationResource,
} as const;
export type CreateOrganization = typeof CreateOrganization;
